frontend/packages/metal3-plugin/src/plugin.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  Plugin,
+  ResourceNSNavItem,
+  ResourceListPage,
+  ResourceDetailPage,
+  ModelFeatureFlag,
+  ModelDefinition,
+} from '@console/plugin-sdk';
+
+import { BaremetalHostModel } from './models';
+
+type ConsumedExtensions =
+  | ResourceNSNavItem
+  | ResourceListPage
+  | ResourceDetailPage
+  | ModelFeatureFlag
+  | ModelDefinition;
+
+const METAL3_FLAG = 'METAL3'
+
+const plugin: Plugin<ConsumedExtensions> = [
+  {
+    type: 'ModelDefinition',
+    properties: {
+      models: [BaremetalHostModel],
```

Comment:
You can also do

```diff
-import { BaremetalHostModel } from './models';
+import * as models from './models';
```

```diff
-models: [BaremetalHostModel],
+models: _.values(models),
```

like in #1592 to avoid manually importing individual models (if you expect to have lots of them).

Reply:
I don't expect there to be many more models, so I listed it explicitly here

Reply:
OK, I'm fine with that.